src/integration/java/io/pinecone/helpers/IndexManager.java

```diff
@@ -53,27 +52,30 @@ public static String createNewIndex(Pinecone pinecone, int dimension, String ind
         CreateIndexRequestSpec createIndexRequestSpec;
 
         if (indexType.equalsIgnoreCase(IndexModelSpec.SERIALIZED_NAME_POD)) {
-            CreateIndexRequestSpecPod podSpec = new CreateIndexRequestSpecPod().environment(environment).podType("p1.x1");
-            createIndexRequestSpec = new CreateIndexRequestSpec().pod(podSpec);
+            pinecone.createPodsIndex(indexName, dimension, environment, "p1.x1");
+            if (waitUntilIndexIsReady) {
+                waitUntilIndexIsReady(pinecone, indexName);
+            }
+            return indexName;
         } else {
-            // Serverless currently has limited availability in specific regions, hard-code us-west-2 for now
-            ServerlessSpec serverlessSpec =
-                    new ServerlessSpec().cloud(ServerlessSpec.CloudEnum.AWS).region("us-west-2");
-            createIndexRequestSpec = new CreateIndexRequestSpec().serverless(serverlessSpec);
+            pinecone.createServerlessIndex(indexName, "cosine", dimension, ServerlessSpec.CloudEnum.AWS.toString(), "us-west-2");
+            if (waitUntilIndexIsReady) {
+                waitUntilIndexIsReady(pinecone, indexName);
+            }
+            return indexName;
         }
+    }
 
-        CreateIndexRequest createIndexRequest = new CreateIndexRequest()
-                .name(indexName)
-                .dimension(dimension)
-                .metric(IndexMetric.DOTPRODUCT) // Sparse-dense is only supported with DOTPRODUCT
-                .spec(createIndexRequestSpec);
-        pinecone.createIndex(createIndexRequest);
+    public static Pinecone createNewIndex(Pinecone pinecone, String indexName, int dimension,
+     String metric, boolean waitUntilIndexIsReady) throws InterruptedException,
+            PineconeException {
+//        CreateIndexRequest createIndexRequest = new CreateIndexRequest().name(indexName).dimension(dimension).metric(metric).spec(spec);
```

Comment:
nit: Comment can come out, I'm also removing all of this `IndexManager` stuff in a follow up so not a huge deal.

Reply:
Oopsy poopsy that shouldn't have been there, ty!